spyderplugins/widgets/pylintgui.py

```diff
@@ -52,10 +51,14 @@ def get_pylint_version():
     global PYLINT_PATH
     if PYLINT_PATH is None:
         return
-    process = subprocess.Popen([PYLINT, '--version'],
-                               stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                               cwd=osp.dirname(PYLINT_PATH),
-                               shell=True if os.name == 'nt' else False)
+    kwargs=dict(cwd=osp.dirname(PYLINT_PATH))
+    cmd = PYLINT
+    args = ['--version']
+    if os.name == 'nt':
+        cmd = PYLINT + ' '.join(args)
+        process = programs.run_shell_command(cmd, **kwargs)
+    else:
+        process = programs.run_program(cmd, args, **kwargs)

```

Comment:
It would be clearer to define cmd and args in each branch. `cmd` is redefined in the first branch, and the join seems pointless since args is defined just before as a list with one value.

Also I think it lacks a space between `PYLINT` and `' '.join(args)`.